lib/rubocop/cop/rails/deprecated_active_model_errors_methods.rb

```diff
@@ -121,23 +121,19 @@ def on_send(node)
         def autocorrect(corrector, node)
           receiver = node.receiver
 
-          if receiver.receiver.send_type? && receiver.receiver.method?(:messages)
-            corrector.remove(receiver.receiver.loc.dot)
-            corrector.remove(receiver.receiver.loc.selector)
-          end
-
           range = offense_range(node, receiver)
           replacement = replacement(node, receiver)
 
           corrector.replace(range, replacement)
         end
 
         def offense_range(node, receiver)
-          range_between(receiver.receiver.source_range.end_pos, node.source_range.end_pos)
+          receiver = receiver.receiver while receiver.send_type? && !receiver.method?(:errors) && receiver.receiver
```

Comment:
Technically, for the current patterns, `!receiver.method?(:errors)` would be enough, but it seemed like a good idea to do the other checks in case new patterns are added. We don't want an infinite loop or NoMethodErrors.